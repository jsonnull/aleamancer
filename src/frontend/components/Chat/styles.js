// @flow
import styled from 'styled-components'

export const Container = styled.div`
  background-color: ${props => props.theme.background};
  width: 320px;
  overflow: hidden;
  border-radius: ${props => (props.isPinned ? '0' : '5px')};
  box-shadow: ${props =>
    props.isPinned ? '' : '0 3px 5px 0 rgba(0, 0, 0, 0.2)'};

  height: ${props => (props.isPinned ? '100%' : 'auto')};
  position: ${props => (props.isPinned ? 'relative' : 'absolute')};
  top: ${props => (props.isPinned ? '0' : 'auto')};
  right: ${props => (props.isPinned ? '0' : '1rem')};
  bottom: ${props => (props.isPinned ? '0' : '1rem')};

  display: flex;
  flex-direction: column;
`

export const Loading = styled.div`
  padding: 10px;
`

export const Error = Loading
