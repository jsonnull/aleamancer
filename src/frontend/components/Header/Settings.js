// @flow
import React from 'react'
import Button from './Button'

type Props = {
  showSettings: Function
}

const Settings = (props: Props) => {
  const { showSettings } = props

  return (
    <Button square onClick={showSettings}>
      <i className="fa fa-cog" />
    </Button>
  )
}

export default Settings
