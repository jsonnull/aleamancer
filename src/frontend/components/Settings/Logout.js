// @flow
import React from 'react'
import styled from 'styled-components'
import Button from '../Button'

type Props = {
  logout: Function
}

const Container = styled.div`
  margin-top: 2rem;
  display: flex;
`

const Logout = (props: Props) => {
  return (
    <Container>
      <Button red onClick={props.logout}>
        Logout
      </Button>
    </Container>
  )
}

export default Logout
