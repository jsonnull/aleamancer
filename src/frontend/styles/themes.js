// @flow
import type { ThemeName } from 'common/types'
import { colors } from './common'

export type Theme = {
  name: ThemeName,
  background: string,
  backgroundSecondary: string,
  backgroundInverted: string,
  backgroundInvertedSecondary: string,
  color: string,
  colorSecondary: string,
  colorMedium: string,
  colorInverted: string,
  map: string,
  borderColor: string,
  modalOverlay: string,
  modalBackground: string
}

export const light: Theme = {
  name: 'light',
  background: '#f7f7f7',
  backgroundSecondary: '#ebe3d6',
  backgroundInverted: '#3c3c3c',
  backgroundInvertedSecondary: '#555',
  color: 'black',
  colorSecondary: '#3c3c3c',
  colorMedium: '#999',
  colorInverted: '#f7f7f7',
  map: '#bfbca5',
  borderColor: '#cdc5ba',
  modalBackground: '#f7f7f7',
  modalOverlay: 'rgba(0, 0, 0, 0.5)'
}

export const dark: Theme = {
  name: 'dark',
  background: '#3c3c3c',
  backgroundSecondary: '#2b2b2b',
  backgroundInverted: colors.blue,
  backgroundInvertedSecondary: colors.lightBlue,
  color: '#eee',
  colorSecondary: '#cdc5ba',
  colorMedium: '#999',
  colorInverted: '#eee',
  map: '#1a1a1a',
  borderColor: '#1a1a1a',
  modalBackground: '#2b2b2b',
  modalOverlay: 'rgba(0, 0, 0, 0.7)'
}
