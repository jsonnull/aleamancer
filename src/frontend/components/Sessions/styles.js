// @flow
import styled from 'styled-components'
import { fontSize, fonts } from 'frontend/styles/common'

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: stretch;
  flex: 1;
`

export const Body = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
  background: ${props => props.theme.background};
  padding: 0 2rem;
`

export const Heading = styled.h1`
  font-size: ${fontSize.large};
  line-height: 1;
  font-family: ${fonts.heading};
  padding: 1rem 0;
  color: ${props => props.theme.color};
  margin: 0;
`
