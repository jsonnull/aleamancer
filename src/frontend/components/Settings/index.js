// @flow
import React from 'react'
import Modal from '../Modal'
import type { Theme } from 'frontend/styles/themes'
import type { DispatchProps, StateProps } from 'frontend/containers/Settings'
import Logout from './Logout'
import Name from './Name'
import ThemeSwitcher from './ThemeSwitcher'

type Props = {
  theme: Theme
} & StateProps &
  DispatchProps

const Settings = (props: Props) => {
  const {
    showSettings,
    displayName,
    changeDisplayName,
    theme,
    changeTheme,
    logout,
    dismissSettings
  } = props

  return (
    <Modal show={showSettings} dismiss={dismissSettings}>
      {() => (
        <div>
          <Name name={displayName} onChange={changeDisplayName} />

          <ThemeSwitcher currentTheme={theme.name} changeTheme={changeTheme} />

          <Logout logout={logout} />
        </div>
      )}
    </Modal>
  )
}

export default Settings
