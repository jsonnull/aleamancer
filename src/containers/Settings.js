// @flow
import React from 'react'
import { connect } from 'react-redux'
import Settings from 'components/Settings'
import { changeTheme, changeDisplayName } from 'actions'
import * as themes from 'styles/themes'
import type { ThemeName } from 'types'
import type { Theme } from 'styles/themes'
import type { State } from 'store'

export type StateProps = {
  theme: Theme,
  displayName: string,
  showSettings: boolean
}
const mapStateToProps = (state: State): StateProps => ({
  theme: themes[state.user.preferences.theme],
  displayName: state.user.profile.displayName,
  showSettings: state.ui.showSettings
})

export type DispatchProps = {
  changeTheme: (theme: ThemeName) => void,
  changeDisplayName: (name: string) => void,
  logout: Function,
  dismissSettings: Function
}
const mapDispatchToProps = (dispatch: Function): DispatchProps => ({
  changeTheme: (theme: ThemeName) => dispatch(changeTheme(theme)),
  changeDisplayName: (name: string) => dispatch(changeDisplayName(name)),
  logout: () => dispatch({ type: 'PERFORM_USER_LOGOUT' }),
  dismissSettings: () => dispatch({ type: 'HIDE_SETTINGS' })
})

export default connect(mapStateToProps, mapDispatchToProps)(Settings)
