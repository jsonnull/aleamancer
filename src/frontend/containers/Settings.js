// @flow
import { connect } from 'react-redux'
import Settings from 'frontend/components/Settings'
import { changeDisplayName, changeTheme } from 'frontend/actions'
import { HIDE_SETTINGS, PERFORM_USER_LOGOUT } from 'frontend/actions/types'
import displayNameSelector from 'frontend/selectors/displayName'
import * as themes from 'frontend/styles/themes'
import type { ThemeName } from 'common/types'
import type { Theme } from 'frontend/styles/themes'
import type { State } from 'frontend/store'

export type StateProps = {
  theme: Theme,
  displayName: string,
  showSettings: boolean
}
const mapStateToProps = (state: State): StateProps => ({
  theme: themes[state.preferences.theme],
  displayName: displayNameSelector(state),
  showSettings: state.ui.showSettings
})

export type DispatchProps = {
  changeTheme: (theme: ThemeName) => void,
  changeDisplayName: (name: string) => void,
  logout: Function,
  dismissSettings: Function
}
const mapDispatchToProps = (dispatch: Function) => ({
  changeTheme: (theme: ThemeName) => dispatch(changeTheme(theme)),
  changeDisplayName: (name: string) => dispatch(changeDisplayName(name)),
  logout: () => dispatch({ type: PERFORM_USER_LOGOUT }),
  dismissSettings: () => dispatch({ type: HIDE_SETTINGS })
})

export default connect(mapStateToProps, mapDispatchToProps)(Settings)
