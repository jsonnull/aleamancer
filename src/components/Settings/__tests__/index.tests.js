// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import Settings from '../index.js'
import { light } from '../../../styles/themes'

describe('Settings component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Settings
          showSettings={true}
          displayName="test"
          changeDisplayName={() => {}}
          theme={light}
          changeTheme={() => {}}
          logout={() => {}}
          dismissSettings={() => {}}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
