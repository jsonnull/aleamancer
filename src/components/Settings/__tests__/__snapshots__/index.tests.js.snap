// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings component renders correctly 1`] = `
<div
  className="sc-bdVaJa hGAKWv"
>
  <div
    className="sc-bwzfXH kELLZF"
  >
    <div>
      <div>
        <label
          className="sc-ifAKCX cmPjpt"
        >
          Display Name:
        </label>
        <div
          className="sc-gzVnrw csXZdG"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          test
        </div>
      </div>
      <div>
        <label
          className="sc-ifAKCX cmPjpt"
        >
          Theme
        </label>
        <div
          className="sc-iwsKbI SIqHi"
        >
          <button
            className="sc-htpNat eKDpsT"
            onClick={[Function]}
          >
            Light
          </button>
          <button
            className="sc-dnqmqq eRYZVa"
            onClick={[Function]}
          >
            Dark
          </button>
        </div>
      </div>
      <div
        className="sc-bxivhb bIqXCY"
      >
        <button
          className="sc-htpNat bnmVIn"
          onClick={[Function]}
        >
          Logout
        </button>
      </div>
    </div>
  </div>
</div>
`;
